Odata-docs/webapi-8/fundamentals/actions-functions.md

```diff
@@ -0,0 +1,423 @@
+
+Side effects means that actions results in a change in the server; either a resource is updated, deleted or a new resource is created.
+
+## Bound and unbound operations.
+From OData V4 spec, functions and actions can be either bound to a type or unbound.
+Bound operations are bound to an entity type, primitive type, complex type, or a collection.
+Unbound operations are invoked as static operations on the service.
+
+## URI patterns for calling OData actions and functions
+In the examples below `odata` is the route prefix.
+
+### Bound function
+The function is after an entity, entityset, singleton. `Books` and `Graphs` are entity sets.
+
+```
+GET ~/odata/Books/mostRecent()
+GET ~/odata/Graphs/Default.GetShapeCount()
+GET ~/odata/Graphs/Default.GetShapeCount(shapeType=FunctionActionBlog.ShapeType’Circle’)
+```
+
+`GetShapeCount` function has an overload that accepts a `ShapeType` parameter.
+
+We can bind to a single entity.
+
+`GET ~/odata/Books(1)/isChildBook()`
+
+We can also bind to a singleton
+
+`GET ~/odata/Me/isMyCalendarBlocked()`
+
+### Unbound function
+Function is after the route prefix.
+
+```
+GET ~/odata/ReturnAllForKidsBooks
+GET ~/odata/GetSalesTaxRate(10)
+```
+`ReturnAllForKidsBooks` doesn't have a parameter, while `GetSalesTaxRate` accepts one integer parameter.
+
+If `GetSalesTaxRate` had more than one parameter, the call would be
+
+`GET ~/odata/GetSalesTaxRate(10, "US")`
+
+Parameter-less functions can work with or without parenthesis `()` after function names.
+This is how we configure parameter-less functions to work without parenthesis.
+
+```csharp
+builder.Services.AddControllers().AddOData(opt =>
+{
+    opt.AddRouteComponents("odata", EdmModel.GetEdmModel()).Count().OrderBy().Filter().Select().Expand();
+    opt.RouteOptions.EnableNonParenthesisForEmptyParameterFunction = true;
+});
+```
+
+### Bound Action
+Actions are invoked via `POST` requests therefore we can pass a payload in the body.
+
+```json
+POST ~/odata/Books('1')/Rate
+{
+    "rating": 7
+}
+```
+
+### Unbound action
+```json
+POST ~/odata/incrementBookYear
+{
+    "increment": 7,
+    "id": "1"
+}
+```
+
+## Practical examples
+We will add detailed examples on how to configure bound and unbound functions and actions.
+We will start by setting up the ASP.NET Core OData v8.x project.
+
+### Data Model classes
+```csharp
+public class Book
+{
+    public string ID { get; set; }
+    public string Title { get; set; }
+    public bool ForKids { get; set; }
+}
+
+public class BookRating
+{
+    public string ID { get; set; }
+    public int Rating { get; set; }
+    public string BookID { get; set; }
+}
+```
+
+### Controller
+```csharp
+public class BooksController : ODataController
+{
+    // Get ~/Books
+    [EnableQuery]
+    public IActionResult Get()
+    {
+        return Ok(DataSource.Instance.Books);
+    }
+}
+```
+
+### Build Edm Model
+```csharp
+private static IEdmModel GetEdmModel()
+{
+    var builder = new ODataConventionModelBuilder();
```

Comment:
`ODataConventionModelBuilder()` is in a separate NuGet package: `Microsoft.OData.ModelBuilder`. Maybe that's worth mentioning? Or maybe that'll be covered in the "Getting Started" guide?

Reply:
Model builder should not be covered in this section